### PR TITLE
8037384: Fix wording in Javadoc of java.io.Serializable

### PR DESCRIPTION
**Pull request by stuart-marks**
8231547: Serializable class doc should link to serialization specification

Rewrite a couple confusing sentences in the Serializable class doc. This does affect normative text, but the edits are primarily to focus and clarify the text, not to make any semantic changes. Thus, a CSR request shouldn't be required for this change.

Also add and adjust some links and link markup to the Java Object Serialization Specification.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux aarch64 | Linux arm | Linux ppc64le | Linux s390x | Linux x64 | Linux x86 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- | ----- | ----- | ----- | ----- |
| Build | ✔️ (1/1 passed) | ✔️ (1/1 passed) | ✔️ (1/1 passed) | ✔️ (1/1 passed) | ✔️ (6/6 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) |    |     |     |     |  ✔️ (9/9 passed) | ❌ (1/9 failed) | ❌ (1/9 failed) | ✔️ (9/9 passed) |

**Failed test tasks**
- [Linux x86 (hs/tier1 compiler)](https://github.com/stuart-marks/jdk/runs/1427734589)
- [Windows x64 (langtools/tier1)](https://github.com/stuart-marks/jdk/runs/1427909089)

### Issues
 * [JDK-8037384](https://bugs.openjdk.java.net/browse/JDK-8037384): Fix wording in Javadoc of java.io.Serializable
 * [JDK-8231547](https://bugs.openjdk.java.net/browse/JDK-8231547): Serializable class doc should link to serialization specification

### Reviewers
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**) ⚠️ Review applies to ce7e6d6c36e97dceab67f0fe74d82aa92c46de45
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**) ⚠️ Review applies to ce7e6d6c36e97dceab67f0fe74d82aa92c46de45
 * [Chris Hegarty](https://openjdk.java.net/census#chegar) (@ChrisHegarty - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1306/head:pull/1306`
`$ git checkout pull/1306`

### CODE REVIEW

**Review by RogerRiggs — APPROVED** [review]

File: src/java.base/share/classes/java/io/Serializable.java [hunk 1]
```diff
@@ -43,20 +43,19 @@
...
- * runtime. <p>
- *
- * During deserialization, the fields of non-serializable classes will
- * be initialized using the public or protected no-arg constructor of
- * the class.  A no-arg constructor must be accessible to the subclass
- * that is serializable.  The fields of serializable subclasses will
- * be restored from the stream. <p>
+ * It is possible for subtypes of non-serializable classes to be serialized
+ * and deserialized. During serialization, no data will be written for the
+ * fields of non-serializable classes. During deserialization, the fields of
```
Comment (ChrisHegarty) [inline src/java.base/share/classes/java/io/Serializable.java:48]: " fields of non-serializable classes" - this strictly means " fields of non-serializable SUPERclasses", right?

File: src/java.base/share/classes/java/io/Serializable.java [hunk 2]
```diff
@@ -43,20 +43,19 @@
...
+ * non-serializable classes will be initialized using the no-arg constructor of
```
Comment (ChrisHegarty) [inline src/java.base/share/classes/java/io/Serializable.java:49]: non-serializable SUPERclasses ..

File: src/java.base/share/classes/java/io/Serializable.java [hunk 3]
```diff
@@ -43,20 +43,19 @@
...
+ * the class. This constructor must be accessible to the subclass that is being
```
Comment (ChrisHegarty) [inline src/java.base/share/classes/java/io/Serializable.java:50]: .. of the first non-serializable superclass.

**Review by irisclark — APPROVED** [review]
Adjusted references look good.

File: src/java.base/share/classes/java/io/Serializable.java (hunk 3 above)
Comment (stuart-marks) [reply src/java.base/share/classes/java/io/Serializable.java]: I updated these to say "superclasses" plus a bit of other minor adjustment.

**Review by ChrisHegarty — APPROVED** [review]
LGTM.

**Pull request closed without merging**